### ISSUE TITLE
Avoid unnecessary JVM forking

### ISSUE THREAD

**Issue by lukebemish** [lukebemishprojects/TaskGraphRunner#2]
Some thoughts on this:
- The standard "run main class with args" setup in many cases is more than enough -- that is, fork the JVM once, and run everything inside that, and if something kills it restart it. However, some things use `System.exit`, which... yeah, isn't great for us. Thoughts? Maybe run a transformer on the classes before loading that replaces these with `Utils.fakeExit` that uses an exception?
- Possible worries:
  - static field pollution in the JVM?
  - things that happen to provide services consumed by base JVM classes?
  - cost of transforming tools -- needs to be cached (lovely...)
  - Such a transformed thing would actually behave differently than if it used `System.exit` -- the exception could be caught from stuff within the surrounding context. The other cannot. So there are cases where it _could_ behave differently.
  - `System.out` and `System.err` pollution -- that's gonna be fun...

So in other words, this isn't something I'd be able to do for _all_ tools. But I should be able to audit a tool, then see if we can get this working with it. So, basically, probably a new `TaskModel`, not just the existing `Tool` task type, that takes a classpath and args and runs on a daemon JVM if one is started, and starts one to run on if the other one is dead. To transform mod jars to remove `System.exit`, it'll need its own caching setup -- to be safe, I'm thinking file name and hash being the necessary part.
